kong/runloop/balancer/targets.lua

```diff
@@ -157,7 +167,14 @@ function targets_M.on_target_event(operation, target)
   log(DEBUG, "target ", operation, " for upstream ", upstream_id,
     upstream_name and " (" .. upstream_name ..")" or "")
 
-  kong.core_cache:invalidate_local("balancer:targets:" .. upstream_id)
+  targets_by_upstream_id["balancer:targets:" .. upstream_id] = nil
+
+  local upstream = upstreams.get_upstream_by_id(upstream_id)
+  if not upstream then
+    log(ERR, "target ", operation, ": upstream not found for ", upstream_id,
+      upstream_name and " (" .. upstream_name ..")" or "")
+    return
+  end
```

Comment:
This code appears to be a duplicate of [this block below](https://github.com/Kong/kong/blob/0b0fb6a0615adb9d5a04cd388b64b559c4ff3e5b/kong/runloop/balancer/targets.lua#L190-L195). Can the latter be removed?

Reply:
🤦 
I missed that when resolving the cherry-pick conflict. Thank you!